source/custom/xnetsocket_converters.h

```diff
@@ -563,6 +567,17 @@ inline AddrInfoHintInputConverter convert_from_grpc(const AddrInfoHint& input)
   return AddrInfoHintInputConverter(input);
 }
 
+inline void convert_to_addr_info_flags(int32_t flags, pb_::RepeatedField<int32_t>& get_addr_info_flags)
+{
+  int flag_to_check = GetAddrInfoFlags::GET_ADDR_INFO_FLAGS_BYPASS_CACHE;
+  while (flag_to_check != 0) {
+    if (flags & flag_to_check) {
+      get_addr_info_flags.Add(flag_to_check);
```

Comment:
I'm thinking we should check `GetAddrInfoFlags_IsValid` here as well. In some scenario where there is an extra "flag we don't know about" I think it's better that we only include it in `_raw` instead of forcing it in here.

Reply:
Agreed. Technically, I don't think we would've hit that scenario since we're starting from the max value of the `GetAddrInfoFlags` enum and bit shifting down through them and only adding them (any ones in raw we don't know about wouldn't be found iterating through the `GetAddrInfoFlags`). I did add it for sanity's sake, though. I guess in general part of this is trusting that `GetAddrInfoFlags` is a proper Flags enum (each element is a power of 2) so bit shifting through them is valid but I think that's a fair assumption to make.

Also, I found protoc generates a `_MAX` constant for enums so I replaced `flag_to_check` with `GetAddrInfoFlags_MAX` instead so it should be future-proof if more flags get added (unlikely).

Reply:
Gotcha, I missed that. Still a good update though.